src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerJsonProcessor.java

```diff
@@ -206,13 +209,14 @@ public Content buildProviderJson(final Repository repository,
       pkg.put(VERSION_KEY, version);
       pkg.put(DIST_KEY, dist);
       pkg.put(TIME_KEY, time);
-      pkg.put(UID_KEY, Hashing.md5().newHasher()
-          .putString(vendor, StandardCharsets.UTF_8)
-          .putString(project, StandardCharsets.UTF_8)
-          .putString(version, StandardCharsets.UTF_8)
-          .putString(time, StandardCharsets.UTF_8)
-          .hash()
-          .asInt());
+      pkg.put(UID_KEY, Integer.toUnsignedLong(
```

Comment:
Question: What was the reason for this change?

Reply:
> Question: What was the reason for this change?

It's somewhat of a long story, but basically I didn't want to have negative values for UIDs.